tctl "clusters" command to control reverse tunnels

kontsevoy opened the pull request:
It always existed as undocumentd 'rts' command.
Making it more user friendly and visible

Refs #309

kontsevoy committed:
tctl "clusters" command to control reverse tunnels

It always existed as undocumentd 'rts' command.
Making it more user friendly and visible

Refs #309

kontsevoy commented:
Mm... looks like "clusters" command should be more high-level. This one only creates a reverse tunnel, but how would the cluster trust you?

klizhentas commented:
@kontsevoy I think we should have a `clusters` command that uses two lower level commands:

reverse tunnels and public authorities for trust - this would constitute the clusters command, while the lower level commands can be hidden by default and used in some special cases

kontsevoy committed:
Configuration changes

1. data_dir is now a global setting in teleport.yaml (instead of being
   inside of "storage" sub-section)

2. changing data_dir in one place causes all of teleport to use it,
   not just bolt backends.

3. moving auth server to listen on non-default ports properly adjusts
   the global auth_servers setting

4. `tctl` now accepts -c flag just like Teleport, so you can pass
   `teleprot.yaml` to it.

Fixes #432
Fixes #431
Fixes #430
